### DISCUSSION TITLE
DropdownMenu Item triggered on Enter (when opening the dropdown)

### DISCUSSION THREAD

**Discussion by fsteenkamp** [radix-ui/primitives#1337] [Help] [answered]
Hi,

I'm trying to compose DropdownMenuItem with DialogTrigger. But the DialogTrigger get's triggered when opening the Dropdown with the 'Enter' key.

Minimal Example:
``` js
<DropdownMenu.Root>
  <DropdownMenu.Trigger>Open</DropdownMenu.Trigger>
  <DropdownMenu.Content>
    <DropdownMenu.Item asChild>
      <button onClick={() => console.log("1")}>Click Me 1</button>
    </DropdownMenu.Item>
    <DropdownMenu.Item asChild>
      <button onClick={() => console.log("2")}>Click Me 2</button>
    </DropdownMenu.Item>
  </DropdownMenu.Content>
</DropdownMenu.Root>
```

When hitten 'Enter' on the dropdown trigger, the console printout is "1" which is unexpected.

As a side-note, I also see a console error: `onloadwff.js:71 Assertion failed: Input argument is not an HTMLInputElement` when opening the Dropdown with Enter. But this is unrelated to the first DropdownItem being triggered as it also happens without the `asChild` composition.

Hope someone can help, thanks in advance! 😄

**Comment by jjenzz** [comment, marked as answer]
this issue comment might help https://github.com/radix-ui/primitives/issues/1268#issuecomment-1080677112
